ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java

```diff
@@ -196,9 +197,11 @@ String toString(RaftGroupId groupId, CompletableFuture<RaftServerImpl> f) {
   private final ExecutorService executor;
 
   private final JvmPauseMonitor pauseMonitor;
+  @Nullable
```

Comment:
Let's create a default `ThreadGroup` and make it non-nullable.

Reply:
Actually I figure a `null` here is fine. [doc](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#Thread-java.lang.ThreadGroup-java.lang.Runnable-) specifies that is the `ThreadGroup` arg is null, the current thread's group is inherited. That behavior can be better than we guess a default here?

Reply:
Interestingly... OpenJDK does not really respect a `null` ThreadGroup here and throws NPE on me. I will change this to non-null.